ticketutil/servicenow.py

```diff
@@ -0,0 +1,321 @@
+
+        Fields example:
+        'contact_type' : 'Email',
+        'u_opened_for' : 'PNT',
+        'assigned_to' : 'pzubaty',
+        'impact' : '2',
+        'urgency' : '2',
+        'priority' : '2'
+        """
+        if short_description is None:
+            logging.error('short_description '
+                          'is a necessary parameter for ticket creation.')
+            return
+
+        params = self._create_ticket_parameters(short_description, kwargs)
+        self._create_ticket_request(params)
+
+    def _create_ticket_parameters(self,
+                                  short_description,
+                                  fields):
+        """
+        Creates the payload for the POST request when creating new ticket.
+
+        :param short_description: short description of the issue
+
+        :param fields: optional fields
+        """
+        params = '"short_description" : "{}"'.format(short_description)
+        for key, value in fields.items():
+            params += ', "{}" : "{}"'.format(key, value)
+        params = '{' + params + '}'
+        return params
+
+    def _create_ticket_request(self, params):
+        """
+        Tries to create the ticket through the ticketing tool's API.
+        Retrieves the ticket_id and creates the ticket_url.
+
+        :param params: The payload to send in the POST request.
+        """
+        result = self.api_post(self.rest_url, params)
+        self.ticket_id = result['number']
+        self.sys_id = result['sys_id']
+        self.ticket_url = self._generate_ticket_url()
+        logging.info('Created issue {0} - {1}'.format(self.ticket_id,
+                                                      self.ticket_url))
+
+    def change_status(self, status):
+        """
+        Change ServiceNow ticket status
+
+        :param status: State to change to
+        """
+        if not self.sys_id:
+            logging.error('No ticket ID associated with ticket object. Set '
+                          'ticket ID with set_ticket_id(ticket_id)')
+            return
+
+        try:
+            logging.info('Changing ticket status')
+            fields = {'state': status}
+            self.edit(**fields)
+        except:
+            logging.error('Failed to change ticket status')
+
+    def edit(self, **kwargs):
+        """
+        Edit ticket
+
+        Edits a ServiceNow ticket, ticked_id (sys_id) must be set beforehand.
+        You can set ticket_id by calling set_ticket_id(ticket_id) method.
+
+        :param kwargs: optional fields
+
+        Fields example:
+        'contact_type' : 'Email',
+        'u_opened_for' : 'PNT',
+        'assigned_to' : 'pzubaty',
+        'impact' : '2',
+        'urgency' : '2',
+        'priority' : '2'
+        """
+        if not self.sys_id:
+            logging.error('No ticket ID associated with ticket object. Set '
+                          'ticket ID with set_ticket_id(ticket_id)')
+            return
```

Comment:
An error occurred, the user should know about this. If you make the `sys_id` property change as discussed above, this entire condition can be removed.